util/ffmpeg.go

```diff
@@ -0,0 +1,136 @@
+	"strings"
+
+	log "maunium.net/go/maulogger/v2"
+)
+
+var ffmpegDefaultParams = []string{"-hide_banner", "-loglevel", "warning"}
+
+// MimeExtensionSanityOverrides includes extensions for various common mimetypes.
+//
+// This is necessary because sometimes the OS mimetype database and Go interact in weird ways,
+// which causes very obscure extensions to be first in the array for common mimetypes
+// (e.g. image/jpeg -> .jpe, text/plain -> ,v).
+var MimeExtensionSanityOverrides = map[string]string{
+	"image/png":  ".png",
+	"image/webp": ".webp",
+	"image/jpeg": ".jpg",
+	"image/tiff": ".tiff",
+	"image/heif": ".heic",
+	"image/heic": ".heic",
+
+	"audio/mpeg": ".mp3",
+	"audio/ogg":  ".ogg",
+	"audio/webm": ".webm",
+	"video/mp4":  ".mp4",
+	"video/mpeg": ".mpeg",
+	"video/webm": ".webm",
+
+	"text/plain": ".txt",
+	"text/html":  ".html",
+
+	"application/xml": ".xml",
+}
+
+func ExtensionFromMimetype(mimetype string) string {
+	ext, ok := MimeExtensionSanityOverrides[strings.Split(mimetype, ";")[0]]
+	if !ok {
+		exts, _ := mime.ExtensionsByType(mimetype)
+		if len(exts) > 0 {
+			ext = exts[0]
+		}
+	}
+	return ext
+}
+
+// Convert a media file on the disk using ffmpeg.
+//
+// Args:
+// * inputFile: The full path to the file.
+// * outputExtension: The extension that the output file should be.
+// * inputArgs: Arguments to tell ffmpeg how to parse the input file.
+// * outputArgs: Arguments to tell ffmpeg how to convert the file to reach the wanted output.
+// * removeInput: Whether the input file should be removed after converting.
+//
+// Returns: the path to the converted file.
+func ConvertPath(inputFile string, outputExtension string, inputArgs []string, outputArgs []string, removeInput bool) (string, error) {
+	outputFilename := strings.TrimSuffix(inputFile, filepath.Ext(inputFile)) + "." + outputExtension
+
+	args := []string{}
+	args = append(args, ffmpegDefaultParams...)
+	args = append(args, inputArgs...)
+	args = append(args, "-i", inputFile)
+	args = append(args, outputArgs...)
+	args = append(args, outputFilename)
+
+	log.Info(args)
+
+	cmd := exec.Command("ffmpeg", args...)
+	vcLog := log.Sub("ffmpeg").Writer(log.LevelWarn)
+	cmd.Stdout = vcLog
+	cmd.Stderr = vcLog
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("ffmpeg error: %+v. stdout: %s. stderr: %s", err, cmd.Stdout, cmd.Stderr)
+	}
+
+	if removeInput {
+		os.Remove(inputFile)
+	}
+
+	return outputFilename, nil
+}
+
+// Convert media data using ffmpeg.
+//
+// Args:
+// * data: The media data to convert
+// * outputExtension: The extension that the output file should be.
+// * inputArgs: Arguments to tell ffmpeg how to parse the input file.
+// * outputArgs: Arguments to tell ffmpeg how to convert the file to reach the wanted output.
+// * inputMime: The mimetype of the input data.
+//
+// Returns: the converted data
+func ConvertBytes(data []byte, outputExtension string, inputArgs []string, outputArgs []string, inputMime string) ([]byte, error) {
```

Comment:
These should probably either be moved to `util/ffmpeg/ffmpeg.go` so they're used as `ffmpeg.ConvertBytes`, or renamed to `util.ConvertBytesFFmpeg`

(although `ExtensionFromMimetype` doesn't really belong in `util/ffmpeg/` if the convert methods are moved there)